From 2efd19e6e3d43bc97383dcff45c9205bbb1ee9c7 Mon Sep 17 00:00:00 2001
From: Stella Laurenzo <stellaraccident@gmail.com>
Date: Mon, 24 Feb 2025 23:05:42 -0800
Subject: [PATCH 3/3] Depend on libdw from libraries that consume its header.

---
 source/lib/common/CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/source/lib/common/CMakeLists.txt b/source/lib/common/CMakeLists.txt
index bb43bc2..4870156 100644
--- a/source/lib/common/CMakeLists.txt
+++ b/source/lib/common/CMakeLists.txt
@@ -38,6 +38,7 @@ target_link_libraries(
     rocprofiler-sdk-common-library
     PUBLIC $<BUILD_INTERFACE:rocprofiler-sdk::rocprofiler-sdk-headers>
            $<BUILD_INTERFACE:rocprofiler-sdk::rocprofiler-sdk-build-flags>
+           $<BUILD_INTERFACE:rocprofiler-sdk::rocprofiler-sdk-dw>
            $<BUILD_INTERFACE:rocprofiler-sdk::rocprofiler-sdk-threading>
            $<BUILD_INTERFACE:rocprofiler-sdk::rocprofiler-sdk-memcheck>
            $<BUILD_INTERFACE:rocprofiler-sdk::rocprofiler-sdk-cxx-filesystem>
@@ -48,7 +49,8 @@ target_link_libraries(
            $<BUILD_INTERFACE:rocprofiler-sdk::rocprofiler-sdk-ptl>
            $<BUILD_INTERFACE:rocprofiler-sdk::rocprofiler-sdk-atomic>
            $<BUILD_INTERFACE:rocprofiler-sdk::rocprofiler-sdk-hsakmt-nolink>
-           $<BUILD_INTERFACE:rocprofiler-sdk::rocprofiler-sdk-elfio>)
+           $<BUILD_INTERFACE:rocprofiler-sdk::rocprofiler-sdk-elfio>
+           )
 
 set_target_properties(rocprofiler-sdk-common-library PROPERTIES OUTPUT_NAME
                                                                 rocprofiler-sdk-common)
-- 
2.43.0

