diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4e4ad5407..759a16b71 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -47,9 +47,9 @@ if(CLR_BUILD_HIP)
     endif()
     message(STATUS "HIPCC Binary Directory: ${HIPCC_BIN_DIR}")
 
-    if(NOT EXISTS ${HIPCC_BIN_DIR}/hipconfig)
-        message(FATAL_ERROR "Please pass hipcc/build or hipcc/bin using -DHIPCC_BIN_DIR.")
-    endif()
+    # if(NOT EXISTS ${HIPCC_BIN_DIR}/hipconfig)
+    #     message(FATAL_ERROR "Please pass hipcc/build or hipcc/bin using -DHIPCC_BIN_DIR.")
+    # endif()
 
     message(STATUS "HIP Common Directory: ${HIP_COMMON_DIR}")
     if(NOT DEFINED HIP_COMMON_DIR)
