diff --git a/CMakeLists.txt b/CMakeLists.txt
index 02df74d..ceba0f9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -183,8 +183,8 @@ if (NOT HIPIFY_CLANG_TESTS_ONLY)
   if(NOT HIPIFY_INCLUDE_IN_HIP_SDK)
     # Install all folders under clang/version/ in CMAKE_INSTALL_PREFIX path.
     install(
-      DIRECTORY ${LLVM_DIR}/../../clang/${LIB_CLANG_RES}/
-      DESTINATION .
+      DIRECTORY ${LLVM_DIR}/../../clang/${LIB_CLANG_RES}/include/
+      DESTINATION include/hipify
       COMPONENT clang-resource-headers
       FILES_MATCHING
       PATTERN "*.h"
@@ -198,8 +198,8 @@ if (NOT HIPIFY_CLANG_TESTS_ONLY)
 
 # install all folders under clang/version/ in CMAKE_INSTALL_PREFIX path
 install(
-    DIRECTORY ${LLVM_DIR}/../../clang/${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.${LLVM_VERSION_PATCH}/
-    DESTINATION .
+    DIRECTORY ${LLVM_DIR}/../../clang/${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.${LLVM_VERSION_PATCH}/include/
+    DESTINATION include/hipify
     COMPONENT clang-resource-headers
     FILES_MATCHING
     PATTERN "*.h"
diff --git a/packaging/hipify-clang.txt b/packaging/hipify-clang.txt
index 27dae2e..0d75178 100644
--- a/packaging/hipify-clang.txt
+++ b/packaging/hipify-clang.txt
@@ -9,7 +9,7 @@ install(PROGRAMS @HIPIFY_BIN_INSTALL_PATH@/hipconvertinplace-perl.sh DESTINATION
 install(PROGRAMS @HIPIFY_BIN_INSTALL_PATH@/hipconvertinplace.sh DESTINATION @CMAKE_INSTALL_BINDIR@)
 install(PROGRAMS @HIPIFY_BIN_INSTALL_PATH@/hipexamine-perl.sh DESTINATION @CMAKE_INSTALL_BINDIR@)
 install(PROGRAMS @HIPIFY_BIN_INSTALL_PATH@/hipexamine.sh DESTINATION @CMAKE_INSTALL_BINDIR@)
-install(DIRECTORY @CMAKE_INSTALL_PREFIX@/include/ DESTINATION @CMAKE_INSTALL_INCLUDEDIR@/hipify)
+install(DIRECTORY @CMAKE_INSTALL_PREFIX@/include/hipify/ DESTINATION @CMAKE_INSTALL_INCLUDEDIR@/hipify)
 
 set (FILE_REORG_BACKWARD_COMPATIBILITY "@FILE_REORG_BACKWARD_COMPATIBILITY@")
 if(FILE_REORG_BACKWARD_COMPATIBILITY)
