From cdfca7b5c4152be6a57361dda2e294bdd9731979 Mon Sep 17 00:00:00 2001
From: Babak <bpoursar@amd.com>
Date: Wed, 13 Nov 2024 21:13:01 +0000
Subject: [PATCH 1/5] Update CHANGELOG.md for ROCm 6.4.0

---
 CHANGELOG.md | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CHANGELOG.md b/CHANGELOG.md
index fbbe5042..222f20ac 100644
--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,6 @@
 # Changelog for Tensile
 
-## (Unreleased) Tensile 4.43.0
+## Tensile 4.43.0 for ROCm 6.4
 
 ### Added
 
-- 
2.43.0

