From 34fd1ff38cafe1e8455a658f881d16bbc03db2cf Mon Sep 17 00:00:00 2001
From: David Dixon <165835255+ellosel@users.noreply.github.com>
Date: Thu, 30 Jan 2025 13:07:56 -0700
Subject: [PATCH 3/5] Only warn when caps mismatch (#2090) (#2091)

---
 Tensile/Common.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Tensile/Common.py b/Tensile/Common.py
index af5f94d5..410c8365 100644
--- a/Tensile/Common.py
+++ b/Tensile/Common.py
@@ -2148,7 +2148,7 @@ def GetAsmCaps(isaVersion: IsaVersion, compilerVersion: CompilerVersion) -> Dict
       elif derivedAsmCaps != CACHED_ASM_CAPS[isaVersion]:
         exitFlag = True
       if exitFlag:
-        printExit("Cached asm caps differ from derived asm caps for {}".format(isaVersion))
+        printWarning("Cached asm caps differ from derived asm caps for {}".format(isaVersion))
     return derivedAsmCaps
   else:
     printWarning("Assembler not present, asm caps loaded from cache are unverified")
-- 
2.43.0

