From 20b027cca8ce5d1cc816d2ecbf2150ce811cc3f0 Mon Sep 17 00:00:00 2001
From: Scott <scott.todd0@gmail.com>
Date: Fri, 14 Feb 2025 15:22:48 -0800
Subject: [PATCH] Enable MSVC support in comgr.

---
 amd/comgr/CMakeLists.txt | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/amd/comgr/CMakeLists.txt b/amd/comgr/CMakeLists.txt
index ed01d21cea30..e643256f2a89 100644
--- a/amd/comgr/CMakeLists.txt
+++ b/amd/comgr/CMakeLists.txt
@@ -228,7 +228,12 @@ elseif (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
     "/wd4624" #[[Suppress 'derived class' : destructor could not be generated because a base class destructor is inaccessible]]
     "/wd4267" #[[Suppress 'var' : conversion from 'size_t' to 'type', possible loss of data]]
     "/wd4291" #[[Suppress 'declaration' : no matching operator delete found; memory will not be freed if initialization throws an exception]]
-    "/wd4146" #[[Suppress 'unary minus operator applied to unsigned type, result still unsigned]])
+    "/wd4146" #[[Suppress 'unary minus operator applied to unsigned type, result still unsigned]]
+
+    # Enable standards-conforming preprocessor.
+    # https://learn.microsoft.com/en-us/cpp/build/reference/zc-preprocessor
+    "/Zc:preprocessor"
+  )
   list(APPEND AMD_COMGR_PRIVATE_COMPILE_DEFINITIONS _HAS_EXCEPTIONS=0)
 endif()
 
-- 
2.45.1.windows.1

